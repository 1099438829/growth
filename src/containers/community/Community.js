/* eslint-disable class-methods-use-this */
import React, { PureComponent } from 'react';
import { View, Text } from 'react-native';
import { Icon, Divider } from 'react-native-elements';

import Colors from '../../theme/colors';
import ForumContainer from './forum/ForumContainers';

const TimeAgo = require('react-native-timeago');
const moment = require('moment'); // load moment module to set local language
require('moment/locale/zh-cn'); // for import moment local language file during the application build

moment.locale('zh-cn');// set moment local language to zh-cn


class Community extends PureComponent {
  static componentName = 'Community';

  renderElement(topic) {
    let sticky = <View />;
    if (topic.attributes.isSticky) {
      sticky = (<View
        style={{
          marginLeft: 5,
          marginRight: 10,
          backgroundColor: Colors.brand.primary,
          padding: 5,
          borderRadius: 2,
        }}
      >
        <Text style={{ marginTop: -3, color: '#fff' }}>置顶</Text>
      </View>);
    }

    return (<View
      style={{
        height: 72,
        backgroundColor: '#fff',
      }}
    >
      <View style={{ margin: 8 }}>
        <Text numberOfLines={1} style={{ fontSize: 15 }}>{topic.attributes.title}</Text>
      </View>
      <View style={{ flex: 1, flexDirection: 'row', margin: 8, justifyContent: 'space-between' }}>
        <View style={{ flex: 1, flexDirection: 'row' }}>
          {sticky}
          <View><Text style={{ color: '#333' }}><TimeAgo time={topic.attributes.lastTime} interval={20000} /></Text></View>
        </View>
        <View style={{ flexDirection: 'row' }}>
          <Icon style={{ marginLeft: 5, marginRight: 5 }} name={'ios-chatboxes-outline'} type={'ionicon'} color={'#333'} />
          <Text>{topic.attributes.commentsCount}</Text>
        </View>
      </View>
      <Divider style={{ backgroundColor: '#eee' }} />
    </View>
    );
  }

  render() {
    return (
      <ForumContainer url={'https://forum.growth.ren/api/discussions'} element={data => this.renderElement(data)} />
    );
  }
}

export default Community;
