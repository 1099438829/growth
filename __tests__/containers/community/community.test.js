import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Community from '../../../src/containers/community/Community';

describe('test user center', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Community />,
    );

    const treeJson = tree.toJSON();
    expect(treeJson).toMatchSnapshot();
  });
});
