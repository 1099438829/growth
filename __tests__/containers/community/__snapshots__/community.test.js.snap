// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test user center renders correctly 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    collapsable={false}
  >
    <View
      style={
        Object {
          "backgroundColor": "#2196f3",
          "elevation": 4,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0.5,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 0.5,
          "zIndex": 1,
        }
      }
    >
      <View
        pointerEvents="none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffeb3b",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
              "width": 0,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "overflow": "scroll",
          }
        }
      />
    </View>
  </View>
</View>
`;
